lib/features/email/data/network/email_api.dart

```diff
@@ -414,8 +414,19 @@ class EmailAPI with HandleSetErrorMixin {
 
       final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
       final currentSetEmailInvocations = currentExecuteList.map((currentItem) {
+
+        var moveProperties = currentItem.value.generateMapUpdateObjectMoveToMailbox(currentItem.key, moveRequest.destinationMailboxId);
+        if ( moveRequest.moveAction == MoveAction.moving && moveRequest.emailActionType == EmailActionType.moveToSpam){
+          var spamProperties = currentItem.value.generateMapUpdateObjectMarkAsRead(ReadActions.markAsRead);
+          moveProperties.forEach((key, value) {
+            if (spamProperties.containsKey(key)) {
+              value.patches.addAll(spamProperties[key]!.patches);
+            }
+          });
+        }
```

Comment:
You should separate the processing function for it, writing it like this is a bit confusing :)

Reply:
You can create new `PatchObject` for move email to spam.

```dart
PatchObject generateMoveToSpamActionPath(
  MailboxId currentMailboxId,
  MailboxId spamMailboxId
) {
    return PatchObject({
      currentMailboxId.generatePath():  null,
      spamMailboxId.generatePath(): true,
      KeyWordIdentifier.emailSeen.generatePath(): true
    });
  }
```

Reply:
@dab246 i fixed it, i created new PatchObject and also new function list_email_id_extension.dart, but i have a question, is generateMapUpdateObjectMarkAsSpam even used ? Because i did not find that it is used.

Reply:
> @dab246 i fixed it, i created new PatchObject and also new function in list_email_id_extension.dart for using the newly created PatchObject, but i have a question, is generateMapUpdateObjectMarkAsSpam even used ? Because i did not find that it is used.

Oh, maybe it's unused from later edited versions. Please delete it for us. Than

Reply:
@dab246 it is deleted and code in email_api.dart is imporved, i tried to squash the commits into one but i dont think i improved it, when this PR will be merged the GitHub can squash it or not ?